platform/website/src/components/auth/auth-dialog.svelte

```diff
@@ -109,201 +93,58 @@
-		id: "email",
-		type: "email",
-		label: "Email",
-		autoComplete: "email",
-		reload() {
-			email.touched = email.value.length > 0;
-			if (email.touched) {
-				email.update(email.value);
-			}
-		},
-		update(value) {
-			email.value = value;
-
-			if ($authDialog !== AuthDialog.Register) {
-				return;
-			}
-
-			const valid = z.string().email("Please enter a valid email").safeParse(email.value);
-			if (!valid.success) {
-				email.status = "error";
-				email.message = valid.error.issues[0].message;
-				return;
-			}
-
-			email.status = "success";
-			email.message = "";
-		},
-		valid() {
-			const status = email.status as string;
-			return status === "success";
-		},
-	});
-
-	let password = newField({
-		id: "password",
-		type: "password",
-		label: "Password",
-		reload() {
-			password.touched = password.value.length > 0;
-			password.status = "";
-			password.message = "";
-
-			password.autoComplete =
-				$authDialog === AuthDialog.Login ? "current-password" : "new-password";
-
-			password.update(password.value);
-		},
-		update(value) {
-			password.value = value;
-
-			// When we are in login mode, we dont need to validate the password.
-			if ($authDialog !== AuthDialog.Register) {
-				return;
-			}
-
-			// Since the validity of confirm password depends on the password, we need to cause a re-render of the confirm password field.
-			confirmPassword.update(confirmPassword.value);
-
-			const valid = z
-				.string()
-				.min(8, "At least 8 characters")
-				.max(100, "Maximum of 100 characters")
-				.regex(/.*[A-Z].*/, "At least one uppercase character")
-				.regex(/.*[a-z].*/, "At least one lowercase character")
-				.regex(/.*\d.*/, "At least one number")
-				.regex(/.*[`~<>?,./!@#$%^&*()\-_+="'|{}[\];:].*/, "At least one special character")
-				.safeParse(value);
-
-			if (!valid.success) {
-				password.status = "error";
-				password.message = valid.error.issues[0].message;
-				return;
-			}
-
-			password.status = "success";
-			password.message = "";
-		},
-		valid() {
-			const value = password.value as string;
-			const status = password.status as string;
-
-			if ($authDialog === AuthDialog.Login) {
-				return value.length > 0;
-			}
-
-			return status === "success";
-		},
-	});
-
-	let confirmPassword = newField({
-		id: "confirmPassword",
-		type: "password",
-		label: "Confirm Password",
-		autoComplete: "new-password",
-		reload() {
-			confirmPassword.touched = confirmPassword.value.length > 0;
-			confirmPassword.status = "";
-			confirmPassword.message = "";
-		},
-		update(value) {
-			confirmPassword.value = value;
-
-			if ($authDialog !== AuthDialog.Register) {
-				return;
-			}
+		});
+	}
 
-			if (password.value !== confirmPassword.value) {
-				confirmPassword.status = "error";
-				confirmPassword.message = "Passwords do not match";
-				return;
-			}
+	let emailStatus: FieldStatus;
+	let emailValue: string;
+	async function emailValidate(v: string): Promise<FieldStatus> {
```

Comment:
Do async functions need to annotate that their resp type is Promise in TypeScript?

Reply:
No, no idea why I did that LULE